design/airgap-existing-cluster.md

```diff
@@ -0,0 +1,110 @@
+# Support for installing KOTS Admin Console in Airgapped Clusters
+
+## Goals
+
+- Install kotsadm to an existing cluster where the cluster does not have internet access
+- Install kotsadm to an existing cluster from a workstation where the workstation does not have internet access at the time of installation
+- Support different credentials for push / pull (workstation / cluster) to the local registry
+- Support temporary/dynamic and persistent/static credentials for pushing kotsadm images to the registry
+- Support only persistent/static credentials for pulling kotsadm images to the registry
+- Support different credentials for pushing vs pulling images to the local registry
```

Comment:
This seems redundant with 


```
- Support different credentials for push / pull (workstation / cluster) to the local registry
```

is there a difference between these?

Reply:
Yeah, there's a difference and this part is necessary, although we can make it clearer. Pushing is a one-time action, so you can use temporary credentials to do it, but pulling can happen any time later, as Kubernetes needs to pull in images, so the credentials need to be long-lived. But even with that, it's possible to design a system such that both credentials need to be the same because we only accept the credentials once. We consciously designed it such that you could specify different credentials.